src/bh_route_blocks.erl

```diff
@@ -69,8 +69,13 @@ get_block_list([{before, undefined}, {limit, Limit}]) ->
     {ok, _, Results} = ?PREPARED_QUERY(?S_BLOCK_LIST, [Limit]),
     {ok, block_list_to_json(Results)};
 get_block_list([{before, Before}, {limit, Limit}]) ->
-    {ok, _, Results} = ?PREPARED_QUERY(?S_BLOCK_LIST_BEFORE, [Before, Limit]),
-    {ok, block_list_to_json(Results)}.
+    case catch binary_to_integer(Before) of
```

Comment:
Use try?

Reply:
stylistic? Or is there a deeper reason?

Reply:
saves on a pattern match and a term conversion?